src/gl/index_buffer.js

```diff
@@ -21,14 +22,22 @@ class IndexBuffer {
         }
 
         gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, this.buffer);
-        gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, array.arrayBuffer, gl.STATIC_DRAW);
-        delete array.arrayBuffer;
+        gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, array.arrayBuffer, this.dynamicDraw ? gl.DYNAMIC_DRAW : gl.STATIC_DRAW);
+
+        if (!this.dynamicDraw) {
+            delete array.arrayBuffer;
+        }
     }
 
     bind() {
         this.gl.bindBuffer(this.gl.ELEMENT_ARRAY_BUFFER, this.buffer);
     }
 
+    updateData(array: TriangleIndexArray | LineIndexArray) {
+        this.bind();
```

Comment:
Does it make sense to assert here that it's in `dynamicDraw` mode?

Reply:
yep!